code/game/machinery/computer/buildandrepair.dm

```diff
@@ -378,19 +378,23 @@
 				return 1
 			if(iswelder(P))
 				var/obj/item/weapon/weldingtool/WT = P
-				if(!WT.remove_fuel(0, user))
-					to_chat(user, "The welding tool must be on to complete this task.")
+				if(WT.remove_fuel(0, user))
+					to_chat(user, "<span class='notice'>You start welding the frame back into metal.</span>")
+					if(do_after(user, src, 10) && state == 0)
+						if(!src || !WT.isOn())
+							return
+						playsound(get_turf(src), 'sound/items/Welder.ogg', 50, 1)
+						user.visible_message("[user] welds the frame back into metal.", "You weld the frame back into metal.", "You hear welding.")
+						var/obj/item/stack/sheet/metal/M = getFromPool(/obj/item/stack/sheet/metal, src.loc)
+						M.amount = 5
+						state = -1
+						qdel(src)
+					else
+						to_chat(user, "<span class='notice'>The welding tool needs to be on to start this task.</span>")
```

Comment:
Turns out the piece of code I mindlessly copypasted from is wrong as well. I'll fix it Later™